packages/catalog-process/test/catalogService.integration.test.ts

```diff
@@ -1547,16 +1702,75 @@ describe("database test", async () => {
           createdAt: new Date(
             Number(writtenPayload.eService?.descriptors[0].createdAt)
           ),
-          docs: [expectedDocument],
+          docs: [expectedDocument1, expectedDocument2],
         };
+
+        const expectedLocalDate = cloneTimeStamp.toLocaleDateString("it-IT");
+        const expectedLocalTime = cloneTimeStamp.toLocaleTimeString("it-IT");
         const expectedEService: EService = {
           ...eService,
           id: unsafeBrandId(writtenPayload.eService!.id),
-          name: `${eService.name} - clone - 1/1/2024 12:00:00`,
+          name: `${eService.name} - clone - ${expectedLocalDate} ${expectedLocalTime}`,
```

Comment:
Could we revert this change? It was specifically written in that way to have a clean overview of the expected formatting.

Reply:
Yes but in my local environment, it makes the test fail. I think that not in every environment `.toLocaleDateString("it-IT")` produces that result.

Reply:
It also causes a problem with `minio`. If we set the backend timezone then we get this error
```
- ApiError during EService creation with name eService nam
e - clone - 1/1/2024 12:00:00
+ The difference between the request time and the server's
 time is too large.
 ```

Reply:
Yes, unfortunately reverting the change breaks the integration tests with MinIO - and it's not straightforward to change the date of the MinIO container to set it to the same date we set here. 
If it is not strictly functional for the test, I would prefer keeping it dynamic like this. Otherwise, I can try to investigate a way to change the date of a single Docker container

Reply:
Based on your points, for me it's not a problem to keep your changes. My concerns were based on [this](https://github.com/pagopa/interop-be-monorepo/pull/174#discussion_r1477983056). Let's see if @galales agrees.

Anyways, if I understand correctly, the problem we are raising now is that we have no clear assurance of how the name will be built. For example, in my local environment (and in the pipeline) it is `${eService.name} - clone - 1/1/2024 12:00:00`, whereas in @MalpenZibo's environment is something like `${eService.name} - clone - 01/01/2024 12:00:00` or similar? (In each environment now it passes because the logic builds the string just like it is expected from the tests, but there might be no consistency)

Reply:
Yes, that's one problem :)
The other problem is that MinIO mimics S3 and S3 requires the requester and the server to be in sync time-wise, otherwise it fails with: `S3 Error: The difference between the request time and the current time is too large`.
So, if we set the date for the tests with `vi.setSystemTime(new Date("2024-01-01T12:00:00"));`, we must somehow sync MinIO with that, which is not straightforward (it would mean setting the system time of the container in some way).

Maybe to improve readability of the string format in this test, we could add a comment above?
For example `//  EService Name - clone - 1/1/2024 12:00:00`

Reply:
> Yes, that's one problem :) The other problem is that MinIO mimics S3 and S3 requires the requester and the server to be in sync time-wise, otherwise it fails with: `S3 Error: The difference between the request time and the current time is too large`.
So, if we set the date for the tests with `vi.setSystemTime(new Date("2024-01-01T12:00:00"));`, we must somehow sync MinIO with that, which is not straightforward (it would mean setting the system time of the container in some way).

I know, I got it. For what concerns this problem, for me it's ok to keep the dynamic date.

> Maybe to improve readability of the string format in this test, we could add a comment above? For example `// EService Name - clone - 1/1/2024 12:00:00`

This doesn't solve the inconsistency. The problem is that the expected string is based on the environment (`1/1/2024` or `01/01/2024`). Maybe we should find a way to build the string in the same way for every environment.

Reply:
> Yes but in my local environment, it makes the test fail. I think that not in every environment .toLocaleDateString("it-IT") produces that result.

This is the reason why the test was explicit.
I do not have strong opinions about the format itself (even though I prefer `01/01/2024 12:00:00`), but it should be at least consistent. 
If we cannot achieve consistency with `.toLocaleDateString("it-IT")` then we probably need to build the date in a different way.

In tests we can re-build the date, being able to avoid mocking the time and having issues with MinIO.

One concern: is it me or this test smells flakiness? If timestamp taken from the test and timestamp taken from the logic falls in two different seconds, wouldn't this test fail?

Reply:
> One concern: is it me or this test smells flakiness? If timestamp taken from the test and timestamp taken from the logic falls in two different seconds, wouldn't this test fail?

@galales what happens with the dynamic date creation is the following:
- on my local environment (and in the pipeline) the logic creates the date in `versionA` and the test expects it in `versionA`
- in @MalpenZibo's local environment the logic creates the date in `versionB` and the test expects it in `versionB`

`versionA`: `1/1/2024 12:00:00`
`versionB`: `01/01/2024 12:00:00`

Reply:
yup, that's why I suggested this 
> If we cannot achieve consistency with .toLocaleDateString("it-IT") then we probably need to build the date in a different way.

meaning that we can avoid relying on that function, let's use something where we can explicitly define the format (which to me seems both clear and consistent)

Reply:
Roger, I will try to explicitly define the formatted date so that we do not rely on `toLocaleDateString("it-IT")`

Reply:
Roger, I will try to explicitly define the formatted date so that we do not rely on toLocaleDateString("it-IT").

For what concerns the test flakiness, I would use a regex at least on the seconds, so that we do not check for an exact second match. Thoughts?